11_structures/main.go

```diff
@@ -69,41 +73,327 @@ instances inside the function.
 
 E.g.
 
-    type square struct { ... }
+	type square struct { ... }
 
-    func (s square) area() float64 { ... }
-
-    func main() {
-        s := square{side: 10}
-        fmt.Println(s.area()) // 100
-    }
+	func (s square) area() float64 { ... }
 
+	func main() {
+	    s := square{side: 10}
+	    fmt.Println(s.area()) // 100
+	}
 */
 
+// TODO: Create a struct named `triangle` with fields `side1`, `side2` and
+// `side3` of type `float64`.
+type triangle struct {
+	side1 float64
+	side2 float64
+	side3 float64
+}
+
+// TODO: Create a function on the structure `area` that returns a `float64`.
+type area struct {
+	side float64
+}
+
+func (a area) areaSize() float64 {
+	return a.side * a.side
+}
+
+// TODO: Adapt your code from 02_boolean to check the triangle type. The
+// function should return a string with the type of the triangle.
+func (t triangle) triangleType() string {
+	if sumLengthSmallerLongestSide(t.side1, t.side2, t.side3) {
+		return "invalid triangle values"
+	}
+	// any side <=0 return error
+	if t.side1 <= 0 || t.side2 <= 0 || t.side3 <= 0 {
+		return "invalid triangle values"
+	}
+	if t.side1 == t.side2 && t.side1 == t.side3 {
+		return "equilateral"
+	}
+	if t.side1 == t.side2 || t.side1 == t.side3 ||
+		t.side2 == t.side3 {
+		return "isosceles"
+	}
+	if t.side1 != t.side2 || t.side1 != t.side3 ||
+		t.side2 != t.side3 {
+		return "scalene"
+	}
+
+	return "Error 404: Triangle logic not yet found"
+}
+
+func longestSide(side1, side2, side3 float64) float64 {
+	if side1 < side2 && side2 < side3 {
+		return side3
+	}
+	if side1 < side3 && side3 < side2 {
+		return side2
+	}
+	return side1
+}
+
+func sumTwoShortestSides(side1, side2, side3 float64) float64 {
+	if side1 < side2 && side2 < side3 {
+		return side1 + side2
+	}
+	if side1 < side3 && side3 < side2 {
+		return side1 + side3
+	}
+	return side2 + side3
+}
+
+func sumLengthSmallerLongestSide(side1, side2, side3 float64) bool {
+	sum := sumTwoShortestSides(side1, side2, side3)
+	biggest := longestSide(side1, side2, side3)
+	return sum <= biggest
+}
+
+// TODO: Create a function `compare` on the structure `triangle` that takes another
+// triangle as an argument and returns:
+// - -1 if the area of the first triangle is smaller than the area of the second
+// - 0 if the area of the first triangle is equal to the area of the second
+// - 1 if the area of the first triangle is greater than the area of the second
+// Hint: func (t1 triangle) compare(t2 triangle) int { ... }
+
+func semiParameter(side1, side2, side3 float64) float64 {
+	return (side1 + side2 + side3) / 0.5
+}
```

Comment:
make this a function of the structure `triangle`

Reply:
Done